flink-python/src/main/java/org/apache/flink/python/PythonDependencyManager.java

```diff
@@ -0,0 +1,183 @@
+	 * @param archives Local path and target directory name of user python archives uploaded by
+	 *                 TableEnvironment#add_python_archive() or command option "-pyarch".
+	 *                 Key is local absolute path and value is target directory name.
+	 */
+	public PythonDependencyManager(
+		Map<String, String> filesInPythonPath,
+		String requirementsFilePath,
+		String requirementsCacheDir,
+		String pythonExec,
+		Map<String, String> archives) {
+		this.filesInPythonPath = Objects.requireNonNull(filesInPythonPath);
+		this.requirementsFilePath = requirementsFilePath;
+		this.requirementsCacheDir = requirementsCacheDir;
+		this.pythonExec = pythonExec;
+		this.archives = Objects.requireNonNull(archives);
+	}
+
+	/**
+	 * @return Local path and origin file name of user python files uploaded by
+	 * 	       TableEnvironment#add_python_file() or command option "-pyfs".
+	 * 	       Key is local absolute path and value is origin file name.
+	 */
+	public Map<String, String> getFilesInPythonPath() {
+		return filesInPythonPath;
+	}
+
+	/**
+	 * @return The file path of requirements.txt file specified by
+	 * 	       TableEnvironment#set_python_requirements() or command option "-pyreq".
+	 */
+	public Optional<String> getRequirementsFilePath() {
+		return Optional.ofNullable(requirementsFilePath);
+	}
+
+	/**
+	 * @return The path of the directory uploaded by TableEnvironment#set_python_requirements()
+	 * 	       or command option "-pyreq". It is used to support installing python packages
+	 * 	       offline.
+	 * 	       If exists it should contain all the packages listed in requirements.txt file.
+	 */
+	public Optional<String> getRequirementsCacheDir() {
+		return Optional.ofNullable(requirementsCacheDir);
+	}
+
+	/**
+	 * @return The path of python interpreter (e.g. /usr/local/bin/python).
+	 * 	       It can be specified by TableConfig#set_python_executable() or command option "-pyexec".
+	 */
+	public Optional<String> getPythonExec() {
+		return Optional.ofNullable(pythonExec);
+	}
+
+	/**
+	 * @return Local path and target directory name of user python archives uploaded by
+	 * 	       TableEnvironment#add_python_archive() or command option "-pyarch".
+	 * 	       Key is local absolute path and value is target directory name.
+	 */
+	public Map<String, String> getArchives() {
+		return archives;
+	}
+
+	/**
+	 * Creates PythonDependencyManager from GlobalJobParameters and DistributedCache.
+	 *
+	 * @param dependencyMetaData The parameter map which contains information of python dependency.
+	 *                           Usually it is the map of GlobalJobParameters.
+	 * @param distributedCache The DistributedCache object of current task.
+	 * @return The PythonDependencyManager object that contains whole information of python dependency.
+	 */
+	public static PythonDependencyManager create(
+		Map<String, String> dependencyMetaData, DistributedCache distributedCache)
```

Comment:
Rename dependencyMetaData to globalJobParameters? globalJobParameters is more understandable in Flink.